src/mapping/query/function/mod.rs

```diff
@@ -120,6 +122,70 @@ build_signatures! {
     contains => ContainsFn,
     slice => SliceFn,
     tokenize => TokenizeFn,
+    split => SplitFn,
+}
+
+pub(in crate::mapping) struct TypedArgument {
+    pub resolver: Box<dyn Function>,
+    pub parameter: Parameter,
+}
+
+impl TypedArgument {
+    pub fn new(resolver: Box<dyn Function>, parameter: Parameter) -> Self {
+        Self {
+            resolver,
+            parameter,
+        }
+    }
+}
+
+// delegates to resolver to satisfy tests in `mapping::parser`.
+impl std::fmt::Debug for TypedArgument {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.resolver.fmt(f)
+    }
+}
+
+impl Function for TypedArgument {
+    fn execute(&self, ctx: &Event) -> Result<Value> {
+        let value = self.resolver.execute(ctx)?;
+
+        // Ask the parameter if it accepts the given value.
+        if !(self.parameter.accepts)(&value) {
+            return Err(format!(
+                "invalid argument type '{}' for parameter '{}'",
+                value.kind(),
+                self.parameter.keyword
+            ));
+        }
+
+        Ok(value)
+    }
+}
+
+/// Because the Regex object doesn't contain a global flag
+/// (this is determined by calling either replace or replace_all)
+/// we need to wrap the object and store this flag when we parse it.
+#[derive(Debug, Clone)]
+pub(in crate::mapping) struct RemapRegex {
+    pub regex: regex::Regex,
+    pub global: bool,
+}
```

Comment:
I'm not sure `RemapRegex` is the best name for this, as on my first glance I wondered what it was remapping. I suggest `RegexArgument` similar to `DynamicArgument` above.